src/UniformBearingCorrector.cpp

```diff
@@ -0,0 +1,172 @@
+
+#include <sonar_driver/UniformBearingCorrector.h>
+
+
+UniformBearingCorrector::UniformBearingCorrector(const UniformBearingCorrectorConfig& config) : config_(config) {
+    computeRemapMatrices(mapX_, mapY_);
+    computeUniformRemapMatrices(uniformMapX_, uniformMapY_);
+}
+
+
+
+
+std::vector<double> UniformBearingCorrector::linspace(const int& start_in, const int& end_in, const int& num_in){
+    std::vector<double> linspaced;
+    double start = static_cast<double>(start_in);
+    double end = static_cast<double>(end_in);
+    double num = static_cast<double>(num_in);
+
+    if (num == 0) { return linspaced; }
+    if (num == 1) {
+      linspaced.push_back(start);
+      return linspaced;
+    }
+
+    double delta = (end - start) / (num - 1);
+  
+    for(int i=0; i < num-1; ++i)    {
+      linspaced.push_back(start + delta * i);
+    }
+    linspaced.push_back(end); // I want to ensure that start and end
+                              // are exactly the same as the input
+    return linspaced;
+}
+
+/// @brief Interpolates the values of a Matrix with known column spacing xp to another column spacing x
+/// @param xp 
+/// @param yp 
+/// @param x 
+/// @return 
+cv::Mat UniformBearingCorrector::interp(const std::vector<int16_t>& xp, const cv::Mat& yp, const std::vector<double>& x){
+    int N_x = x.size();
+    int N_xp = xp.size();
+	cv::Mat y(yp.size(), CV_8UC1, cv::Scalar(0));
+
+	int ip = 0;
+	int ip_next = 1;
+	int i = 0;
+
+	while(i < N_x){
+		double m = static_cast<double>(yp.at<u_int8_t>(0, ip_next)-yp.at<u_int8_t>(0, ip))/(static_cast<double>(xp[ip_next]-xp[ip])/100.0);
+		double q = static_cast<double>(yp.at<u_int8_t>(0, ip)) - m * static_cast<double>(xp[ip])/100.0;
+		
+        while(x[i] < xp[ip_next]){
+			if(x[i] >= xp[ip]){
+                y.at<u_int8_t>(0, i) = static_cast<u_int8_t>(std::round(m*x[i]/100.0+q));
+
+			}
+			i +=1;
+			if (i >= N_x) {break;}
+		}
+
+    	ip +=1;
+		ip_next +=1;
+		if(ip_next == N_xp){
+			while(i < N_x){
+				y.at<u_int8_t>(0, i) = 0;
+				i++;
+			}
+			break;
+		}
+	}
+
+	return y;
+}
```

Comment:
Challenge: Choose worse names for your variables. This is an incomprehensible mess and absolutely unmaintainable. Please use pronounceable names that communicate intent. Have a look at this [resource](https://dev.to/gervaisamoah/a-guide-to-clean-code-the-power-of-good-names-3f6i#:~:text=Principles%20of%20Naming%20in%20Clean%20Code) for more guidelines. And please don't reply with "This is a mathematical method and these are the variables as used in the math", we write code, not math.